### ISSUE TITLE
Make Ggml a standalone package rather than a library in alpaca

### ISSUE THREAD

**Issue by shamblett** [shamblett/alpaca#5]
It may be worth making ggml into its own package if there is enough call for this.

**Comment by abcnow** [comment]
I think you should Steve.. there is a huge need to get something into flutter that will allow to chat with local documents.. take a look at this repo:

https://github.com/PromtEngineer/localGPT

**Comment by shamblett** [comment]
I agree, its just a massive undertaking, your talking around 16000 lines of C code, OK you could start by implementing just the part Alpaca needs but this is still a major effort. 

This s why I looked at trying to automate this somehow, see this Medium [article](https://levelup.gitconnected.com/porting-code-to-dart-using-starcoder-an-evaluation-728423a151f6?sk=a1fea27e0ca55e44450008fa05e398cf)

I haven't reached the stage where I can do this in large enough amounts to be useful, it takes a lot of (expensive) resource to do this in bulk, even then it would have to be checked of course.

**Comment by abcnow** [comment]
I could help with the automatic porting.. I know flutter though my background is in graphic design -no knowledge of C or C++

**Comment by abcnow** [comment]
I read your article.. pretty cool.. I used chatgpt3.5 and the converted code is pretty similar to StarCoder.. the only difference is that "chatgpt" uses "int" instead of "var".. the 1600 lines of code i could do it in 2-4 weeks.. 
---
chatgpt
```

void highPassFilter(List<double> data, double cutoff, double sampleRate) {
  final double rc = 1.0 / (2.0 * 3.14159265359 * cutoff);
  final double dt = 1.0 / sampleRate;
  final double alpha = dt / (rc + dt);
  
  double y = data[0];
  
  for (int i = 1; i < data.length; i++) {
    y = alpha * (y + data[i] - data[i - 1]);
    data[i] = y;
  }
}

```

**Comment by shamblett** [comment]
Ok but its approx 16000 lines as I said above not 1600 this is just a line count from Kate so it will be less but there are other considerations like using simd for vector maths, using isolates maybe for threading etc.

Couple that with a unit test suite and you have quite a project.

I would like to do this but I dont have the time at the mo to do.it in any sane timeframe.
